.github/workflows/main.yml

```diff
@@ -131,7 +131,104 @@ jobs:
+      #
+      # Using `flutter drive` as workaround is not an option because it's super
+      # flaky in terms of stability. With `flutter drive` ~ 30% the Android
+      # Emulator have difficulties to start properly. 
+      - name: Build Android App
+        working-directory: ./app
+        env:
+          USER_1_EMAIL: ${{ secrets.INTEGRATION_TEST_USER_1_EMAIL }}
+          USER_1_PASSWORD: ${{ secrets.INTEGRATION_TEST_USER_1_PASSWORD }}
+        run: flutter build apk --target=integration_test/app_test.dart --flavor dev --dart-define USER_1_EMAIL=$USER_1_EMAIL --dart-define USER_1_PASSWORD=$USER_1_PASSWORD
+      
+      - name: Run integration tests
+        uses: reactivecircus/android-emulator-runner@v2
+        env:
+          USER_1_EMAIL: ${{ secrets.INTEGRATION_TEST_USER_1_EMAIL }}
+          USER_1_PASSWORD: ${{ secrets.INTEGRATION_TEST_USER_1_PASSWORD }}
+        with:
+          # When you are going to change something here (like a different
+          # target, arch, profile, channel, etc.), please check if these
+          # configurations are stable.
+          #
+          # You can do this by running this workflow multiple times. Use a
+          # matrix to run multiple tests in parallel (matrix needs to be copied
+          # to above the "with"):
+          # strategy:
+          #   fail-fast: false
+          #   matrix:
+          #     test1: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+          #     test2: [1, 2, 3]
+          api-level: 30
+          target: google_apis
+          arch: x86_64
+          profile: Nexus 5X
+          channel: canary
+          # We use some emulator options for the following reasons:
+          # * "-no-boot-anim": Disables the boot animation for faster booting.
+          # * "-no-window": Disables the graphical windows, so no display is
+          #   required.
+          # * "-no-snapshot": Disables the quick boot feature. Therefore, the
+          #   emulator does not load or save the emulator state. We want to have
+          #   a fresh testing environment for every test run.
+          # * "-camera-back virtualscene": Shows this virtual room when opening
+          #   the back camera where you can walk. Default option for emulators
+          #   created by Android Studio. We use this so we have the same
+          #   environment locally as when using this action. Is needed when
+          #   testing things with the camera. It's also possible to inject
+          #   images into the virtual scene to test things like qr code scanner.
+          # * "-camera-front": Shows this green monster when opening the front
+          #   camera. Default option for emulators created by Android Studio. We
+          #   use this so we have the same environment locally as when using
+          #   this action. Is needed when testing things with camera.
+          emulator-options: -no-snapshot -no-window -no-boot-anim -camera-back virtualscene -camera-front emulated
```

Comment:
Teeny tiny nit: Could you use the same order for the `emulator-options:` flags and for the comment above explaining them?

Reply:
Changed